### ISSUE TITLE
mm.data().create_image() function does not work

### ISSUE THREAD

**Issue by nicost** [micro-manager/pycro-manager#157]
Not sure if this is a bug or a question of how to use the function, however, the following code snippet illustrates the problem.  Make sure to have an image open in MM and then run this code.

```
from pycromanager import Bridge

bridge = Bridge()
mm = bridge.get_studio()

dv = mm.displays().get_active_data_viewer()
ds = dv.get_datastore()
cb = mm.data().get_coords_builder()
coord = cb.c(0).t(0).p(0).z(0).build()
img = ds.get_image(coord)

out_ds = mm.data().create_ram_datastore()
new_metadata = img.get_metadata().copy_builder_with_new_uuid().build()
new_image = mm.data().create_image(img.get_raw_pixels(), img.get_width(), img.get_height(), 2, 1, coord, new_metadata)
out_ds.put_image(new_image)
mm.displays().createDisplay(out_ds)
```

I get the following error when calling the create_image function:

```
Traceback (most recent call last):
  File "C:\Users\NicoLocal\anaconda3\envs\jupyterN2V\lib\site-packages\IPython\core\interactiveshell.py", line 3417, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-5-b4f70baadbcc>", line 3, in <module>
    new_image = mm.data().create_image(img.get_raw_pixels(), img.get_width(), img.get_height(), 2, 1, coord, new_metadata)
  File "C:\Users\NicoLocal\anaconda3\envs\jupyterN2V\lib\site-packages\pycromanager\core.py", line 280, in <lambda>
    fn = lambda instance, *args, signatures_list=tuple(methods_with_name): instance._translate_call(signatures_list, args)
  File "C:\Users\NicoLocal\anaconda3\envs\jupyterN2V\lib\site-packages\pycromanager\core.py", line 360, in _translate_call
    valid_method_spec = _check_method_args(method_specs, fn_args)
  File "C:\Users\NicoLocal\anaconda3\envs\jupyterN2V\lib\site-packages\pycromanager\core.py", line 486, in _check_method_args
    if _check_single_method_spec(method_spec, fn_args):
  File "C:\Users\NicoLocal\anaconda3\envs\jupyterN2V\lib\site-packages\pycromanager\core.py", line 467, in _check_single_method_spec
    if _ARRAY_TYPE_TO_NUMPY_DTYPE[arg_java_type] != arg_val.dtype:
KeyError: 'java.lang.Object'
```

I am assuming that there is some magic somewhere to convert the Java Object "raw_pixels" into a numpy array, but that this magic is missing in the return path (presumably because I am the first to use it?).  It would be really nice (and for some projects essential) to be able to use the standard MM viewer, so if this is indeed a bug, it would be great to fix it.  Happy to help, although that would take me a lot of digging time;)

**Comment by henrypinkard** [comment]
That sounds right for the error you're seeing, but I am not able to reproduce. On that same line I get:

```
Traceback (most recent call last):
  File "/Users/henrypinkard/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3331, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-309fa94982ab>", line 1, in <module>
    new_image = mm.data().create_image(img.get_pixels(), img.get_width(), img.get_height(), 2, 1, coord, new_metadata)
AttributeError: 'org_micromanager_data_internal_DefaultImage' object has no attribute 'get_pixels'
```

And the source code confirms that `DefaultImage` does not have `get_pixels` (https://github.com/micro-manager/micro-manager/blob/master/mmstudio/src/main/java/org/micromanager/data/internal/DefaultImage.java)

Thoughts on what is different? I am just opening MM, snapping an image, and running python script

**Comment by nicost** [comment]
Sorry, typo.  Should be get_raw_pixels().  Edited above for clarity.

**Comment by henrypinkard** [comment]
Had to add in some special logic for converting numpy arrays to java arrays when passing in as an argument. Should be fixed by https://github.com/micro-manager/micro-manager/pull/976 and available in pycromanager 0.5.0

**Comment by nicost** [comment]
Awesome! Thanks @henrypinkard !
